driver_test.go

```diff
@@ -12,26 +12,263 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package spannerdriver_test
+package spannerdriver
 
 import (
+	"cloud.google.com/go/spanner"
+	"context"
 	"database/sql"
+	"fmt"
 	"log"
+	"os"
+	"reflect"
+	"runtime/debug"
+	"strings"
+	"testing"
 
-	spannerdriver "github.com/rakyll/go-sql-driver-spanner"
+	// API/lib packages not imported by driver.
+	adminapi "cloud.google.com/go/spanner/admin/database/apiv1"
 	"google.golang.org/api/option"
+	adminpb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+	"google.golang.org/grpc"
 )
 
-func ExampleDriver() {
-	driver := &spannerdriver.Driver{
-		Options: []option.ClientOption{
-			option.WithCredentialsFile("/path/to/service-account-key.json"),
-		},
+var (
+	dsn string
+)
+
+type Connector struct {
+	ctx         context.Context
+	client      *spanner.Client
+	adminClient *adminapi.DatabaseAdminClient
+}
+
+func NewConnector() (*Connector, error) {
+
+	// Configure emulator if set.
+	spannerHost, ok := os.LookupEnv("SPANNER_EMULATOR_HOST")
+
+	ctx := context.Background()
+
+	var adminClient *adminapi.DatabaseAdminClient
+	var err error
+
+	if ok {
```

Comment:
Could we extract the `if / else` block alongside the `spannerHost, ok := os.LookupEnv("SPANNER_EMULATOR_HOST")` into its own method for encapsulation? Something like `createAdminClient`.

Reply:
Moved to CreateAdminClient function